lib/active_model/serializer/adapter/nested_json.rb

```diff
@@ -0,0 +1,57 @@
+module ActiveModel
+  class Serializer
+    class Adapter
+      class NestedJson < Json
+        cattr_accessor :default_limit_depth
+        @@default_limit_depth = 5
+
+        def serializable_hash options = {}
```

Comment:
Yikes, it's a pretty big smell in the Json serializer that we're subclassing it but completely overriding this method, where everything important happens

also, might this belong in the Adapter superclass? It looks on first inspection like it would apply to the JsonApi adapter as well, no?